src/core/Akka.API.Tests/CoreAPISpec.ApproveCore.verified.txt

```diff
@@ -1486,6 +1510,16 @@ namespace Akka.Actor
             public static readonly Akka.Actor.ProviderSelection.Remote Instance;
         }
     }
+    public sealed class Reason
+    {
+        public Reason(string type, string message = null) { }
+        public string Message { get; }
+        public string Type { get; }
+    }
```

Comment:
The `public guid Error` property is missing. 
It would make it possible to tag and serialize/log exceptions into elasticsearch, seq, opentelemetry, Redis or simply into a S3 bucket and only transmit this `Reason` to remote nodes. 
If needed the receiving remote system could try restore/deserialize the exception.

Reply:
Maybe instead of `Guid Error` a property `string Url` would be better. 
It will allow to have many/mix of error service types.

Reply:
Why not use a log entry for this? These are just meant to be simple metric updates

Reply:
The `Reason` is need to transfer unknown Exception types between nodes transparently. 
Has not much to to with the `IActorTelemetryEvent` them self.

`Actor.Remote` creates the assumption that all remoting is transparent.  
But if a node transmit a unknown exception type to a node it fails horrible.

The general idea is to put the `Èxception` into memory cache or distributed storage/cache 
then can then be loaded when really needed,  
to get from Reason => Exception again

Reply:
By design, these events are not designed to be transmitted over the wire - local aggregation only. We don't want this data coupled to remoting, serialization, or persistence. "Simple metrics" is the goal.

Reply:
I am asking only to make the `Reason` type multi purpose that it can be in general used instead of an exception.
The remote transparency has this intrinsic conflict with the `Exception` dotnet type.
The MS guys for `system.runtime.remoting` in framework 1.1 times already discovered it.